write_test.go

```diff
@@ -0,0 +1,132 @@
+package dicom
+
+import (
+	"testing"
+	"os"
+	"encoding/binary"
+
+  "github.com/stretchr/testify/assert"
+  "github.com/suyashkumar/dicom/pkg/dicomio"
+  "github.com/suyashkumar/dicom/pkg/tag"
+	"github.com/suyashkumar/dicom/pkg/uid"
+)
+
+/*
+FURTHER TESTING
+	- Read written values back in and verify Datsets are the same
+	- With 'wild' DICOMs with high variability, read in, write out, read in, and verify
+*/
+
+// TODO clean this function up big time
+func TestWrite(t *testing.T) {
+	location := "fullwrite.dcm"
+	file, err := os.Create(location)
+	assert.Nil(t, err)
+	defer file.Close()
+
+	mediaStorageSOPClassUID, err := newElement(tag.MediaStorageSOPClassUID, []string{"1.2.840.10008.5.1.4.1.1.1.2"})
+	assert.Nil(t, err)
+	mediaStorageSOPInstanceUID, err := newElement(tag.MediaStorageSOPInstanceUID, []string{"1.2.3.4.5.6.7"})
+	assert.Nil(t, err)
+	transferSyntax, err := newElement(tag.TransferSyntaxUID, []string{uid.ImplicitVRLittleEndian})
+	assert.Nil(t, err)
+	patientName, err := newElement(tag.PatientName, []string{"Robin Banks"})
+	assert.Nil(t, err)
+
+	elems := []*Element{
+							 mediaStorageSOPClassUID,
+							 mediaStorageSOPInstanceUID,
+							 transferSyntax,
+							 patientName,
+	}
+
+	ds := &Dataset{Elements: elems}
+
+	err =  Write(file, ds, func (set *writeOptSet){})
+	assert.Nil(t, err)
+
+	// TODO verify that the correct values are written
+}
+
+// TODO clean this function up big time
+func TestWriteFileHeader(t *testing.T) {
+	location := "fileheader.dcm"
```

Comment:
we could consider using iotutil.TempFile: https://golang.org/pkg/io/ioutil/#TempFile